temporian/core/operators/calendar/test/test_calendar_day_of_year.py

```diff
@@ -42,6 +41,26 @@ def test_basic(self):
         result = evset.calendar_day_of_year()
         assertOperatorResult(self, result, expected)
 
+    @parameterized.parameters({"tz": 1}, {"tz": 1.0}, {"tz": "Europe/Brussels"})
```

Comment:
missed this comment before, will add this cases

Reply:
Added these tests and one for checking leap years.

I don't love the fact that the tests are spread between all operators, for instance, I tested the leap year for `day_of_the_month` because it's more convenient.
Given that the calendar operators share most code if the leap year is correct for one of the operators then is correct for all of them, so I think this is OK.